src/main/java/com/tragent/inventory/controllers/SupplierController.java

```diff
@@ -0,0 +1,124 @@
+import com.tragent.inventory.service.SupplierService;
+
+@RestController
+@RequestMapping("api/v1/suppliers")
+public class SupplierController {
+	
+	@Autowired
+	private SupplierService supplierService;
+	
+	/**
+	 * Get all suppliers.
+	 * 
+	 * @return all suppliers in the system
+	 */
+	@RequestMapping(method=RequestMethod.GET, 
+			produces = MediaType.APPLICATION_JSON_VALUE)
+	public ResponseEntity<Collection<Supplier>> getUsers(@RequestParam(value = "email", required = false) String email){
+		
+		Collection<Supplier> suppliers = new ArrayList<Supplier>();
+		if (email != null) {
+			Supplier supplier = supplierService.findByEmail(email);
+			suppliers.add(supplier);
+		} else {
+			Collection<Supplier> allSuppliers = supplierService.findAll();
+			suppliers.addAll(allSuppliers);
+		}
+		return new ResponseEntity<Collection<Supplier>>(suppliers, HttpStatus.OK);
+		
+	}
+	
+	/**
+	 * Get supplier with given supplier id.
+	 * 
+	 * @param id id of the supplier to return
+	 * @return the supplier with given id or 404 if id is not found
+	 */
+	@RequestMapping(value="/{id}",
+			method=RequestMethod.GET, 
+			produces = MediaType.APPLICATION_JSON_VALUE)
+	public ResponseEntity<Supplier> getUserById(@PathVariable("id") Long id){
+		
+		Supplier supplier = supplierService.findById(id);
+		if (supplier == null) {
+			return new ResponseEntity<Supplier>(HttpStatus.NOT_FOUND);
+		}
+		
+		return new ResponseEntity<Supplier>(supplier, HttpStatus.OK);
+		
+	}
+	
+	/**
+	 * Create new supplier
+	 * 
+	 * @param supplier
+	 * @return the created supplier and HttpStatus.CREATED if supplier was successfully created
+	 */
+	@RequestMapping(method=RequestMethod.POST,
+			consumes = MediaType.APPLICATION_JSON_VALUE,
+			produces = MediaType.APPLICATION_JSON_VALUE)
+	public ResponseEntity<Supplier> createSupplier(@RequestBody Supplier supplier){
+		
+		supplier = supplierService.create(supplier);
+		if (supplier == null) {
+			return new ResponseEntity<Supplier>(HttpStatus.FORBIDDEN);
+		}
+		
+		return new ResponseEntity<Supplier>(supplier, HttpStatus.CREATED);
+		
+	}
+	
+	/**
+	 * update supplier's information
+	 * 
+	 * @param id, id of the supplier
+	 * @return the updated supplier information.
+	 */
+	@RequestMapping(value="/{id}",
+			method=RequestMethod.PUT,
+			consumes = MediaType.APPLICATION_JSON_VALUE,
+			produces = MediaType.APPLICATION_JSON_VALUE)
+	public ResponseEntity<Supplier> updateUser(@RequestBody Supplier supplier){
+		
+		supplier = supplierService.update(supplier);
+		if (supplier == null) {
+			return new ResponseEntity<Supplier>(HttpStatus.FORBIDDEN);
+		}
+		
+		return new ResponseEntity<Supplier>(supplier, HttpStatus.OK);
+		
+	}
+	
+	/**
+	 * Delete user's information
+	 * 
+	 * @param id, id of the user
+	 * @return 204, .
+	 */
+	@RequestMapping(value="/{id}",
+			method=RequestMethod.DELETE)
+	public ResponseEntity<Supplier> deleteUser(@PathVariable("id") Long id){
+		
+		supplierService.delete(id);
```

Comment:
Do we really want to actually delete suppliers and others, is it better to deactivate in the background instead of really deleting, we may want to see a history of previous suppliers in the future. What do u think?